File: app/src/main/java/com/wayfair/themoviedb/feature/homepage/HomepageRepository.java
package com.wayfair.themoviedb.feature.homepage;

import com.pubbix.base.BaseRepository;

import javax.inject.Inject;

public class HomepageRepository extends BaseRepository<HomepageContract.Interactor>
implements HomepageContract.Repository{
    private HomepageContract.Interactor interactor;
    @Inject
    public HomepageRepository() {
    }

    @Override
    public void setInteractor(HomepageContract.Interactor interactor) {
        this.interactor = interactor;
    }
}

File: app/src/main/java/com/wayfair/themoviedb/feature/homepage/HomepageRepository.kt
package com.wayfair.themoviedb.feature.homepage

import com.pubbix.base.BaseRepository

import javax.inject.Inject

class HomepageRepository @Inject
constructor() : BaseRepository<HomepageContract.Interactor>(), HomepageContract.Repository {
    private var interactor: HomepageContract.Interactor? = null

    override fun setInteractor(interactor: HomepageContract.Interactor) {
        this.interactor = interactor
    }
}
